src/vs/base/common/settingsVerification.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { isObject } from 'vs/base/common/types';
+
+interface IVerifier<T> {
+	verify(value: unknown): T;
+}
+
+abstract class Verifier<T> implements IVerifier<T> {
```

Comment:
How about `verifier.ts`. You call it `settingsVerification.ts` now, but none of the code here is specific to settings.

Alternartively, if we think this is something everyone should be using that uses any kind of configuration, we could also think about having it in the configuration component (e.g. `src/vs/platform/configuration/common/configuration.ts`). I would check with @sandy081